dash_auth/group_protection.py

```diff
@@ -212,3 +296,65 @@ def wrap(*args, **kwargs):
         return wrap
 
     return decorator
+
+
+def protect_layout(pg: dict = None) -> Callable:
```

Comment:
I don't understand the point of this decorator compared to using `@protected` on the layout function?
By default all layouts will need authentication. This would only be used to add specific group-based protection which is unlikely to be uniform across pages.

Reply:
Exactly the point. 😁

Reply:
This also gives the ability to not do any decorations and apply a function for the `protect_layouts` function that can pull the groups based upon the path or path_template, from an external control source.

Reply:
After thinking about it some more, I removed `protect_layout` and utilize `protected` inside of the `protect_layouts` function instead.

This allows for a single `dash.register_page` to house the arguments to get picked up by `protect_layouts` later